src/Microsoft.AspNetCore.Session/SessionMiddleware.cs

```diff
@@ -88,9 +95,9 @@ public class SessionMiddleware
             if (string.IsNullOrWhiteSpace(sessionKey) || sessionKey.Length != SessionKeyLength)
             {
                 // No valid cookie, new session.
-                var guidBytes = new byte[16];
-                CryptoRandom.GetBytes(guidBytes);
-                sessionKey = new Guid(guidBytes).ToString();
+                sessionKey = _sessionKeyGenerator.GetNewSessionKey(SessionKeyLength);
```

Comment:
If anything, the expected session key length would be a property on the ISessionKeyGenerator, not a method parameter. The middleware doesn't need to set the length, only verify it on line 95.